ultralytics/solutions/object_counter.py

```diff
@@ -138,13 +124,19 @@ def mouse_event_for_region(self, event, x, y, flags, params):
             self.is_drawing = False
             self.selected_point = None
 
+    # Define a function to check for intersection
+    @staticmethod
+    def does_intersect(segment1, segment2):
+        """Check if two segments intersect."""
+        return LineString(segment1).intersects(LineString(segment2))
```

Comment:
@RizwanMunawar shouldn't we keep the fixed line as an instance of LineString() so we don't need to keep calling the class?

I would attach it to the class directly to save compute, i.e. `self.line[0] = LineString(segment[0])` and then always use `self.line[0]`.

Reply:
@glenn-jocher yes makes sense. Let me update this. Thanks

Reply:
@glenn-jocher I've made the changes. The line points will be now directly converted to `LineString` in the `init` function, and I've removed the `does_intersect` function since it's no longer needed.